plugins/rest/rest.go

```diff
@@ -45,6 +45,7 @@ type Config struct {
 		OAuth2    *oauth2ClientCredentialsAuthPlugin `json:"oauth2,omitempty"`
 		ClientTLS *clientTLSAuthPlugin               `json:"client_tls,omitempty"`
 		S3Signing *awsSigningAuthPlugin              `json:"s3_signing,omitempty"`
+		Custom    *customAuthPlugin                  `json:"custom,omitempty"`
```

Comment:
nit: I'd name this field Plugin and make it a string. Any additional config would be live under the plugin config.